### PR TITLE
Adds tests for assert x in list

### PR DESCRIPTION
**Pull request by jacqueswww**
### - What I did
I realised we still needed at test reflecting the best use case of  '  in  ' statements :smile_cat: 

### - How to verify it
Check the test.

### CODE REVIEW

**Comment by fubuloubu** [comment]
I would move that assert inside the `set_owner()` function and probably remove the `is_owner()` because you can rewrite the test showing `set_owner()` fails when an owner not in the list tries to add, showing this works more clearly.

**Comment by jacqueswww** [comment]
@fubuloubu please have a look now - I kept is_owner for clarity. But I think it's more realistic now :)

File: tests/parser/features/iteration/test_range_in.py [hunk 1]
```diff
@@ -71,4 +71,33 @@ def testin() -> bool:
...
+    c = get_contract(code)
+
+    assert c.is_owner() is True  # contract creator is owner.
+    assert c.is_owner(sender=t.k1) is False  # no one else is.
+
+    # only an owner may set another owner.
+    with pytest.raises(t.TransactionFailed):
+        c.set_owner(1, t.a1, sender=t.k1)
+
+    assert c.set_owner(1, t.a1) is None
```
Comment (fubuloubu) [inline tests/parser/features/iteration/test_range_in.py]: Unnecessary to write it this way, just change to `c.set_owner(1, t.a1)`

File: tests/parser/features/iteration/test_range_in.py [hunk 2]
```diff
@@ -71,4 +71,33 @@ def testin() -> bool:
...
+    assert c.is_owner(sender=t.k1) is True
```
Comment (fubuloubu) [inline tests/parser/features/iteration/test_range_in.py]: Add another set/assert combo showing a1 can replace a0 in owners list?

File: tests/parser/features/iteration/test_range_in.py [hunk 3]
```diff
@@ -71,4 +71,33 @@ def testin() -> bool:
...
     """
     with pytest.raises(TypeMismatchException):
-        c = get_contract(code)
+        get_contract(code)
+
+
+def test_ownership():
+    code = """
+
+owners: address[10]
```
Comment (fubuloubu) [inline tests/parser/features/iteration/test_range_in.py]: only 2 addresses needed to show functionality below

File: tests/parser/features/iteration/test_range_in.py [hunk 4]
```diff
@@ -71,4 +71,33 @@ def testin() -> bool:
...
+    return msg.sender in self.owners
+    """
+
+    c = get_contract(code)
+
+    assert c.is_owner() is True  # contract creator is owner.
+    assert c.is_owner(sender=t.k1) is False  # no one else is.
+
+    # only an owner may set another owner.
+    with pytest.raises(t.TransactionFailed):
```
Comment (fubuloubu) [inline tests/parser/features/iteration/test_range_in.py]: we have `assert_tx_failed` fixture for this purpose

Comment (fubuloubu) [reply tests/parser/features/iteration/test_range_in.py]: Oh, you need to write it like `assert_tx_failed(t, lambda: c.func() )` for it to work

Comment (jacqueswww) [reply tests/parser/features/iteration/test_range_in.py]: perfect, will make the changes thanks! :)

File: tests/parser/features/iteration/test_range_in.py (hunk 2 above)
Comment (jacqueswww) [reply tests/parser/features/iteration/test_range_in.py]: :+1:

File: tests/parser/features/iteration/test_range_in.py (hunk 3 above)
Comment (jacqueswww) [reply tests/parser/features/iteration/test_range_in.py]: :+1:

File: tests/parser/features/iteration/test_range_in.py (hunk 1 above)
Comment (jacqueswww) [reply tests/parser/features/iteration/test_range_in.py]: :+1:

**Comment by jacqueswww** [comment]
@fubuloubu I replaced the previous commit :smile:  - hope it looks all good now ? :)

**Comment by fubuloubu** [comment]
LGTM

**Comment by fubuloubu** [comment]
Note #446 I updated how `assert_tx_failed()` works so you no longer need to add the `tester` arg

**Comment by jacqueswww** [comment]
@fubuloubu will add when #446 is merged ;)

**Comment by DavidKnott** [comment]
@jacqueswww This looks good, I'll merge it in after you rebase.

**Comment by jacqueswww** [comment]
@DavidKnott done ;)

**Comment by DavidKnott** [comment]
@jacqueswww Thanks a lot!

**Pull request merged**